### DISCUSSION TITLE
fileinfo required by composer.json

### DISCUSSION THREAD

**Discussion by jdwx** [chillerlan/php-qrcode#186] [Ideas]
The composer.json file lists the "fileinfo" extension in the "require" section.

As far as I can tell, fileinfo functions are only used by the Imagick output format.

src/Output/QRImagick.php contains extension_loaded checks for imagick and fileinfo and aborts if either is missing.

The "imagick" extension is **not** required in composer.json.

It seems counterintuitive to require fileinfo but not imagick.  Consistency would suggest including both or neither.

Many (most?) use cases do not involve the Imagick output format, which would seem to favor neither.

Would it make sense to move fileinfo from "require" to "suggest" and add imagick there as well?

**Comment by codemasher** [comment]
Hey, that's a good point! I didn't think about that when i added it. Will change soon!

**Reply by codemasher** [reply]
Fixed in https://github.com/chillerlan/php-qrcode/commit/2b36fa9a088f52027e9995f04ebec8556297b69a
